api/src/main/java/com/basistech/util/TextDomain.java

```diff
@@ -31,30 +32,36 @@
 
     /**
      * Create a TextDomain object.
+     * {@code null} values supplied here are mapped to their corresponding defaults.
      * 
-     * @param script the ISO 15924 numeric script id
-     * @param language the ISO 639 numeric langauge id
-     * @param scheme the TransliterationScheme
+     * @param script the ISO 15924 code. {@code null} is mapped to {@link ISO15924#Zyyy}.
+     * @param language the language code. {@code null} is mapped to {@link LanguageCode#UNKNOWN}.
+     * @param scheme the TransliterationScheme. {@code null} is mapped to {@link TransliterationScheme#UNKNOWN}.
      */
     public TextDomain(ISO15924 script, LanguageCode language, TransliterationScheme scheme) {
-        theScript = script;
-        theLanguage = language;
-        theScheme = scheme;
+        theScript = script == null ? ISO15924.Zyyy : script;
+        theLanguage = language == null ? LanguageCode.UNKNOWN : language;
+        theScheme = scheme == null ? TransliterationScheme.UNKNOWN : scheme;
```

Comment:
The other constructors should probably just be calling the constructor with all params
to avoid this kind of thing, but we still have decide who is right.